test/java/com/rapleaf/jack/test_project/DatabasesImpl.java

```diff
@@ -12,17 +12,21 @@
 import com.rapleaf.jack.test_project.database_1.impl.Database1Impl;
 
 public class DatabasesImpl implements IDatabases {
-  private final IDatabase1 database1;
+  private final BaseDatabaseConnection database1_connection;
+  private IDatabase1 database1;
 
   public DatabasesImpl(BaseDatabaseConnection database1_connection) {
-    this.database1 = new Database1Impl(database1_connection, this);
+    this.database1_connection = database1_connection;
   }
 
   public DatabasesImpl() {
-    this.database1 = new Database1Impl(new DatabaseConnection("database1"), this);
```

Comment:
I don't think this will do what we want though, because new DatabaseConnection("database1") will try to load the db's properties from the yaml.  We need that connection to be lazily instantiated.